### ISSUE TITLE
Change to our git flow: introducing the new stable, build green, `master` branch. 

### ISSUE THREAD

**Issue by mattab** [piwik/piwik#5954]
**Challenge to solve**
Community developers contributing to Piwik (eg. via pull requests) often will checkout a code that has broken tests. Pull requests execute on tests that were already broken so devs get a bad feeling when they start contributing :-1: 

**Solution**
We have discussed this for two hours during our team meetup in Munich and have come to some useful action points:

 * `master` branch should always have a green build!
 * core team members work in `develop` branch
 * we merge `develop` to `master`  when `develop` is green, regularly
 * document our new git flow in [dev docs](http://developer.piwik.org/guides/core-team-workflow)

**Questions**
 * how do we make it work nicely with submodules?
 * how often do we merge `develop` to `master` (daily or every few days?)
 * should we also check that piwik-ui-tests build is green too?
 
Result: anyone issuing Pull requests will have a chance to see green build for their proposed changes, 

this is part of broader improvements to the way Piwik functions internally and as a community. see also #5953 new Changelog.md file for developers

**Comment by mattab** [comment]
In the last few months we have made many improvements to our CI setup.

One major issue remains, it is to fix all randomly failing tests. Luckily we have fixed almost all cases, but here are the last remaining ones:

 *  Some UI screenshot tests fail at random #6693 
 *  System test test_GetCounters_ShouldOnlyReturnResultsOfLastMinutes sometimes fail #7165 

Once we do these fixes, `master` branch will be always green, because Product Engineering team now uses Pull request for everything non-trivial (that could potentially break the build).
